src/bases/finitebasis.jl

```diff
@@ -0,0 +1,60 @@
+import Base: length,
+    size
+
+###############
+# FiniteBasis #
+###############
+    # A FiniteBasis repesents the tensor product
+    # of an arbitrary number of fock bases, each of which is truncated 
+    # at a known, finite number of states. For example, imagine the 
+    # following fock bases and their tensor product:
+    #
+    # F_4 => {|0>, |1>, |2>, |3>}
+    # F_3 => {|0>, |1>, |2>}
+    # F_2 => {|0>, |1>} 
+    #
+    # F_4 x F_3 x F_2 => {|0,0,0>, |0,0,1>, |0,1,0>, |0,1,1>...|3,2,1>}
+    # 
+    # Note that the subscript number refers to basis *length*, not max
+    # excitation level.
+    #
+    # This tensor product basis, F_4 x F_3 x F_2, could be represented 
+    # by calling FiniteBasis(4, 3, 2). 
+
+    immutable FiniteBasis{S} <: AbstractBasis{S}
```

Comment:
You mean instead of `lens::(Int...)` we'd have `lens::NTuple{N,T}`? I'd like to be able to support different lengths for the factor bases - `NTuple` would work for multiple factor bases of uniform length, though, if we wanted to make a separate basis type for that case.